packages/cli-utils/src/index.ts

```diff
@@ -155,8 +138,95 @@ async function main() {
       const shouldPreprocess =
         !options['disable-preprocessing'] && options['disable-preprocessing'] !== 'false';
       return generateTadaTypes(shouldPreprocess);
+    })
+    .command('check')
+    .option('--severity', 'The minimum severity of diagnostics to display.')
+    .action(async (opts) => {
+      const config = await getConfig();
+      if (!config) {
+        return;
+      }
+      const result = (await check(config, opts.severity || 'error')) || [];
+      const errorDiagnostics = result.filter((d) => d.severity === 'error');
+      const warnDiagnostics = result.filter((d) => d.severity === 'warn');
+      const infoDiagnostics = result.filter((d) => d.severity === 'info');
+      if (
+        errorDiagnostics.length === 0 &&
+        warnDiagnostics.length === 0 &&
+        infoDiagnostics.length === 0
+      ) {
+        // eslint-disable-next-line no-console
+        console.log('No issues found! 🎉');
+        return;
+      } else {
+        const errorReport = errorDiagnostics.length
+          ? `Found ${errorDiagnostics.length} Errors:\n\n${constructDiagnosticsPerFile(
+              errorDiagnostics
+            )}\n\n`
+          : ``;
+        const warningsReport = warnDiagnostics.length
+          ? `Found ${warnDiagnostics.length} Warnings:\n\n${constructDiagnosticsPerFile(
+              warnDiagnostics
+            )}\n\n`
+          : ``;
+        const suggestionsReport = infoDiagnostics.length
+          ? `Found ${infoDiagnostics.length} Suggestions:\n\n${constructDiagnosticsPerFile(
+              infoDiagnostics
+            )}\n\n`
+          : ``;
+        // eslint-disable-next-line no-console
+        console.log(`${errorReport}${warningsReport}${suggestionsReport}`);
```

Comment:
(doesn't block merging, can be a todo)

We'd need a summary to separate the error/warn/info list from their respective counts

There's also no `process.exit` codes here yet

Reply:
Added a process.exit, no summary yet as I figured we have no final thoughts on the view yet.